src/App.tsx

```diff
@@ -1,61 +1,139 @@
+import { ChangeEvent, FormEvent, useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+const validationString
+= 'ABCDEFGHIJKLMNOPQRSTUVWXYАБВГҐДЕЄЖЗИІЇЙКЛМНОПРСТУФХЦЧШЩЬЮЯ123456789 ';
+
+type TodoWithUser = Todo & {
+  user: User | null,
+};
+
+const findUserByUserId = (userId: number) => {
+  return usersFromServer.find(({ id }) => id === userId) || null;
+};
+
+const todosWithUser = todosFromServer.map(todo => {
+  const { userId } = todo;
+
+  return {
+    ...todo,
+    user: findUserByUserId(userId),
+  };
+});
 
 export const App = () => {
+  const defaultTitle = '';
+  const defaultUserId = 0;
+
+  const [todos, setTodos] = useState<TodoWithUser[]>(todosWithUser);
+  const [title, setTitle] = useState(defaultTitle);
+  const [userId, setUserId] = useState(defaultUserId);
+  const [titleValid, setTitleValidity] = useState(true);
+  const [userIdValid, setUserIdValidity] = useState(true);
+
+  const getNewId = () => (Math.max(...todos.map(({ id }) => id)) + 1);
```

Comment:
move this function outside of the component
for now this function is recreated on each component rerendering

or let's create it and call only while form submitting action